src/streamlink/plugins/facebook.py

```diff
@@ -35,6 +37,16 @@ def _get_streams(self):
             else:
                 self.logger.debug("Non-dash/mp4 stream: {0}".format(stream_url))
 
+        for match in self._dash_manifest_re.finditer(res.text):
```

Comment:
I wasn't sure what the desired behavior should be here, since the code for handling .mpd links also uses the loop.

I also want to double check that using `streams.update()` and including the existing VOD links is also the correct behavior? The manifest string includes the same quality streams (480 and 720p I think?) named as `sd`/`hd`, so with what I have right there would be two streams for each quality, although `sd` quality would be the 480p .mp4 URL, and the `480p` quality would be for the DASH version (and likewise for the 720p qualities).

Reply:
you probably need `yield` instead of `stream.update()` if the streams get overwritten

example

https://github.com/streamlink/streamlink/blob/87a55fe4a026c5673f4135ddb6a9f03cd9a1c5b4/src/streamlink/plugins/delfi.py#L44-L51

Reply:
`yield` for sure :)

Reply:
One other question before I push any changes for this, if you look at the original code for using a playlist as the fallback when no other streams are found:
https://github.com/streamlink/streamlink/blob/c7386f3d21e49cf9242d01e369d2f8b9fd82ce60/src/streamlink/plugins/facebook.py#L53-L60
both `vod_urls` and `streams` will always be empty if we actually reach this fallback code, since the only place `vod_urls` is modified is:
https://github.com/streamlink/streamlink/blob/c7386f3d21e49cf9242d01e369d2f8b9fd82ce60/src/streamlink/plugins/facebook.py#L34-L36
and hitting that code block means we return `streams` before getting to the playlist fallback block.

So as far as I can tell the fallback loop is just iterating through every video in the playlist so that `streams["sd"]` will eventually contain the last video in the playlist. Should I keep the existing behavior or can the plugin just return the first video in the playlist here?

Reply:
Looks like a bug. I don't remember the specifics of what the playlist fallback was meant to do :)